src/list.lisp

```diff
@@ -323,12 +320,11 @@
   alist)
 
 (defun copy-alist (alist)
-  (let ((new-alist ()))
+  "Return a new association list which is EQUAL to ALIST."
```

Comment:
I like your new implementation better. But note that the order of the functions is not important in the compiler. As long as you don't call it before the other one is defined.  It is okay for a function to use a function that will be defined latter.